src/OpenTelemetry/Metrics/Aggregator/LastValueAggregator.cs

```diff
@@ -0,0 +1,85 @@
+// <copyright file="LastValueAggregator.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Metrics;
+using System.Linq;
+
+namespace OpenTelemetry.Metrics
+{
+    public class LastValueAggregator : Aggregator
+    {
+        private readonly Instrument instrument;
+        private readonly KeyValuePair<string, object>[] tags;
+
+        private readonly object lockUpdate = new object();
+        private int count = 0;
+        private DataPoint lastDataPoint;
+
+        public LastValueAggregator(Instrument instrument, string[] names, object[] values)
+        {
+            this.instrument = instrument;
+
+            if (names.Length != values.Length)
+            {
+                throw new ArgumentException("Length of names[] and values[] must match.");
+            }
+
+            this.tags = new KeyValuePair<string, object>[names.Length];
+            for (int i = 0; i < names.Length; i++)
+            {
+                this.tags[i] = new KeyValuePair<string, object>(names[i], values[i]);
```

Comment:
It's to avoid having another heap alloc in the hot path when we detect a new KVP combination.

TLDR;
We maintain a 2-level dictionary to lookup an aggregator by 1st level = Key[] then 2nd level = Value[]. So, we need to split the KVP into Key[] and Value[] already. The original KVP is on the stack (ReadOnlySpan<>) so we would have to make a copy of that if we used it. But, the Key[] and Value[] needs to be allocated for the Dictionary anyways, so I use that instead of making another copy of the KVP.